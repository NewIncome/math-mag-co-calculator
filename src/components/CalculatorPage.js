/* eslint-disable react/destructuring-assignment */
import React from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

const CalculatorPage = props => (
  <section className="section1">
    <p>Let&apos;s do some math!</p>
    <div id="calcContainer">
      <div id="cBody">
        <div className="calcTop">
          <Display result={
            (props.total === null && props.next === null) ? '0' : props.next === null ? props.total : props.next
            }
          />
        </div>
        <div className="calcBtm">
          <ButtonPanel onClick={props.handleClick} />
        </div>
      </div>
    </div>
  </section>
);

export default CalculatorPage;
