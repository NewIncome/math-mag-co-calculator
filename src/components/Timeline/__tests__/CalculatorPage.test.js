import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import CalculatorPage from '../../CalculatorPage';

it('calculator page renders correctly', () => {
  const tree = renderer.create(<CalculatorPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('calculator page renders exactly all HTML elements', () => {
  const tree = renderer
    .create(<CalculatorPage />)
    .toJSON();
  expect(tree).toMatchSnapshot(`
  <section class="section1">
    <p>Let's do some math!</p>
    <div id="calcContainer">
      <div id="cBody">
        <div class="calcTop">
          <div class="display"></div>
        </div>
        <div class="calcBtm">
          <div class="group1 group"></div>
          <div class="group2 group"></div>
          <div class="group3 group"></div>
          <div class="group4 group"></div>
          <div class="group5 group"></div>
        </div>
      </div>
    </div>
  </section>
  `);
});
