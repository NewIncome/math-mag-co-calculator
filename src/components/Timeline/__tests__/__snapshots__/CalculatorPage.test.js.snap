// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator page renders correctly 1`] = `
<section
  className="section1"
>
  <p>
    Let's do some math!
  </p>
  <div
    id="calcContainer"
  >
    <div
      id="cBody"
    >
      <div
        className="calcTop"
      >
        <div
          className="display"
        />
      </div>
      <div
        className="calcBtm"
      >
        <div
          className="group1 group"
        >
          <button
            className="btn"
            id="A/C"
            onClick={[Function]}
            type="button"
          >
            A/C
          </button>
          <button
            className="btn"
            id="+/-"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="btn"
            id="%"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="btn"
            id="รท"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="group2 group"
        >
          <button
            className="btn"
            id="7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="btn"
            id="8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="btn"
            id="9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="btn"
            id="X"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="group3 group"
        >
          <button
            className="btn"
            id="4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="btn"
            id="5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="btn"
            id="6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="btn"
            id="-"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="group4 group"
        >
          <button
            className="btn"
            id="1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="btn"
            id="2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="btn"
            id="3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="btn"
            id="+"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="group5 group"
        >
          <button
            className="btn"
            id="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="btn"
            id="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="btn"
            id="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`calculator page renders exactly all HTML elements: 
  <section class="section1">
    <p>Let's do some math!</p>
    <div id="calcContainer">
      <div id="cBody">
        <div class="calcTop">
          <div class="display"></div>
        </div>
        <div class="calcBtm">
          <div class="group1 group"></div>
          <div class="group2 group"></div>
          <div class="group3 group"></div>
          <div class="group4 group"></div>
          <div class="group5 group"></div>
        </div>
      </div>
    </div>
  </section>
   1`] = `
<section
  className="section1"
>
  <p>
    Let's do some math!
  </p>
  <div
    id="calcContainer"
  >
    <div
      id="cBody"
    >
      <div
        className="calcTop"
      >
        <div
          className="display"
        />
      </div>
      <div
        className="calcBtm"
      >
        <div
          className="group1 group"
        >
          <button
            className="btn"
            id="A/C"
            onClick={[Function]}
            type="button"
          >
            A/C
          </button>
          <button
            className="btn"
            id="+/-"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="btn"
            id="%"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="btn"
            id="รท"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="group2 group"
        >
          <button
            className="btn"
            id="7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="btn"
            id="8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="btn"
            id="9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="btn"
            id="X"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
        <div
          className="group3 group"
        >
          <button
            className="btn"
            id="4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="btn"
            id="5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="btn"
            id="6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="btn"
            id="-"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="group4 group"
        >
          <button
            className="btn"
            id="1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="btn"
            id="2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="btn"
            id="3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="btn"
            id="+"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="group5 group"
        >
          <button
            className="btn"
            id="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="btn"
            id="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="btn"
            id="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
